lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -440,6 +440,13 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
   DeclName Name = UDRE->getName();
   SourceLoc Loc = UDRE->getLoc();
 
+  // 'Any' is the empty protocol composition; resolve it before doing an
+  // unqualified lookup
+  if (Name.getBaseName() == Context.Id_Any) {
```

Comment:
Ah, is this only for unqualified lookup? That's a little better, but it's still something we should resolve in the parser, not in name lookup. `Any` really is a keyword now, and it should resolve to the type in all contexts. (Though `Any.self` should keep working.)